rageval/metrics/_answer_bleu.py

```diff
@@ -0,0 +1,162 @@
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import datasets
+from datasets import Dataset
+import re
+
+from rageval.metrics import Metric, add_attribute
+
+
+_DESCRIPTION = """\
+BLEU (Bilingual Evaluation Understudy) is an algorithm for evaluating the quality of text which has been machine-translated from one natural language to another.
+Scores are calculated for individual translated segments—generally sentences—by comparing them with a set of good quality reference translations.
+Those scores are then averaged over the whole corpus to reach an estimate of the translation's overall quality.
+Neither intelligibility nor grammatical correctness are not taken into account.
+
+For details, see the paper: http://www.aclweb.org/anthology/P02-1040.pdf
+"""
+
+_KWARGS_DESCRIPTION = """\
+Args:
+    name : str
+    batch_size : int, Batch size for openai completion.
+
+Optional Args:
+    None
+
+Functions:
+    _clean: clean special word in sentence.
+    _compute_single: compute bleu score for single prediction with its references
+
+Examples:
+    >>> from datasets import Dataset
+    >>> import rageval as rl
+    >>> sample = {
+    ...     "answers": [
+    ...         "Hello, world!",
+    ...         "I am a metric named bleu."
+    ...     ],
+    ...     "gt_answers": [
+    ...         [
+    ...             "Hello, my world!",
+    ...         ],
+    ...         [
+    ...             "I am a metric named bleu.",
+    ...             "I am bleu metric.",
+    ...         ]
+    ...     ],
+    ... }
+    >>> dataset = Dataset.from_dict(sample)
+    >>> metric = rl.metrics.AnswerBleuScore()
+    >>> metric.mtype
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset, batch_size=1)
+    >>> assert 0 <= s <= 1
```

Comment:
If the bleuscore of the sample is a unique value, you should `assert s == 0.XXX`, rather than assert a range.